### PR TITLE
Clang fix for Android-S

### PR DESCRIPTION
**Pull request by vdanix**

### CODE REVIEW

File: bsp_diff/caas/hardware/intel/external/drm-intel/0002-Clang-fix-for-drm-intel.patch [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ 				bo_fake->dirty = 0;
+ 
+ 				if (!(bo_fake->flags & BM_NO_FENCE_SUBDATA) &&
+-				    bo_fake->block->fenced) {
++				    ((bo_fake->block !=NULL) && bo_fake->block->fenced)) {
+ 					drm_intel_fake_bo_wait_rendering_locked
+ 					    (bo);
+ 				}
+ 
++				if(bo_fake->block->virtual!=NULL)
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/drm-intel/0002-Clang-fix-for-drm-intel.patch:23]: Add (bo_fake->block !=NULL) also

File: bsp_diff/caas/hardware/intel/external/drm-intel/0002-Clang-fix-for-drm-intel.patch [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ 				bo->virtual = bo_fake->block->virtual;
+ 			}
+ 		} else {
+@@ -1185,7 +1186,8 @@ static int
+ 
+ 	assert(bo_fake->block);
+ 	assert(bo_fake->block->bo == &bo_fake->bo);
+-
++        
++	if (bo_fake->block!=NULL)
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/drm-intel/0002-Clang-fix-for-drm-intel.patch:33]: Can **bo_fake->block->mem** variable be NULL ?

File: bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+diff --git a/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp b/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp
+index 4a6a8c24c..cf1d1c997 100644
+--- a/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp
++++ b/media_driver/agnostic/common/vp/hal/vphal_render_composite.cpp
+@@ -3454,7 +3454,7 @@ int32_t CompositeState::SetLayer(
+                 //For Y210/Y216 with AVS(Y)+3D(U/V) sampler, the shift is not needed.
+                 if ((pSource->Format == Format_Y210 ||
+                     pSource->Format == Format_Y216)
+-                    && pSurfaceEntries[0]->bAVS)
++                    && (pSurfaceEntries[0]->bAVS != NULL && pSurfaceEntries[0]->bAVS))
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch:18]: (pSurfaceEntries[0] != NULL && pSurfaceEntries[0]->bAVS)

File: bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+--- a/media_driver/agnostic/common/vp/hal/vphal_renderer.cpp
++++ b/media_driver/agnostic/common/vp/hal/vphal_renderer.cpp
+@@ -789,8 +789,11 @@ MOS_STATUS VphalRenderer::RenderPass(
+             }
+             // restore render pointer and count.
+             pRenderParams->pTarget[0]            = StoreRenderParams.pTarget[0];
+-            pRenderParams->pTarget[0]->b16UsrPtr = StoreRenderParams.pTarget[0]->b16UsrPtr;
+-            pRenderParams->uDstCount             = StoreRenderParams.uDstCount;
++            
++	    if(StoreRenderParams.pTarget[0]->b16UsrPtr !=NULL)
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch:33]: (pRenderParams->pTarget[0] != NULL && pRenderParams->pTarget[0]->b16UsrPtr !=NULL)

File: bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                     fShiftX = 0.0f;
+                     fShiftY = 0.0f;
+diff --git a/media_driver/agnostic/common/vp/hal/vphal_renderer.cpp b/media_driver/agnostic/common/vp/hal/vphal_renderer.cpp
+index 21b2e8d00..ce0645b8b 100644
+--- a/media_driver/agnostic/common/vp/hal/vphal_renderer.cpp
++++ b/media_driver/agnostic/common/vp/hal/vphal_renderer.cpp
+@@ -789,8 +789,11 @@ MOS_STATUS VphalRenderer::RenderPass(
+             }
+             // restore render pointer and count.
+             pRenderParams->pTarget[0]            = StoreRenderParams.pTarget[0];
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch:29]: if(pRenderParams != NULL) {


}

File: bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+diff --git a/media_driver/agnostic/gen10/vp/hal/vphal_render_composite_g10.cpp b/media_driver/agnostic/gen10/vp/hal/vphal_render_composite_g10.cpp
+index 2ae1e0139..6dfa03f96 100644
+--- a/media_driver/agnostic/gen10/vp/hal/vphal_render_composite_g10.cpp
++++ b/media_driver/agnostic/gen10/vp/hal/vphal_render_composite_g10.cpp
+@@ -55,7 +55,7 @@ void CompositeStateG10::SubmitStatesFillGenSpecificStaticData(
+     // Set ChromaSitting
+     pStatic->DW10.ObjKa2Gen9.ChromaSitingLocation = GetOutputChromaSitting(pTarget);
+ 
+-    if (pRenderingData->iLayers > 0)
++    if (pRenderingData->iLayers > 0 && pRenderingData->pLayers[0]->bIEF !=NULL)
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch:49]: pRenderingData ?

File: bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+diff --git a/media_driver/agnostic/gen9/vp/hal/vphal_render_composite_g9.cpp b/media_driver/agnostic/gen9/vp/hal/vphal_render_composite_g9.cpp
+index ac447bf72..0e6425edb 100644
+--- a/media_driver/agnostic/gen9/vp/hal/vphal_render_composite_g9.cpp
++++ b/media_driver/agnostic/gen9/vp/hal/vphal_render_composite_g9.cpp
+@@ -55,7 +55,7 @@ void CompositeStateG9::SubmitStatesFillGenSpecificStaticData(
+     // Set ChromaSitting
+     pStatic->DW10.ObjKa2Gen9.ChromaSitingLocation = GetOutputChromaSitting(pTarget);
+ 
+-    if (pRenderingData->iLayers > 0)
++    if (pRenderingData->iLayers > 0 && pRenderingData->pLayers[0]->bIEF != NULL)
```
Comment (bnkoppad) [inline bsp_diff/caas/hardware/intel/external/media/mod/0001-Clang-fix-for-media_driver.patch:88]: pRenderingData ? check at all places

**Review by bnkoppad — CHANGES_REQUESTED** [review]

**Review by feijiang1 — APPROVED** [review]
LGTM
But need add Tracked-On jira number

**Comment by vdanix** [comment]
> LGTM But need add Tracked-On jira number

Tracked-On added

**Review by feijiang1 — APPROVED** [review]
LGTM

**Review by bnkoppad — APPROVED** [review]

**Comment by vdanix** [comment]
@basilchew please review the pr

**Comment by basilchew** [comment]
Is there a corresponding PR or Jira for horizontal?

**Comment by vdanix** [comment]
> Is there a corresponding PR or Jira for horizontal?

Horizontal jira : OAM-111281
Vertical revert jira : OAM-111282

**Review by basilchew — APPROVED** [review]

**Pull request merged**